benchmarks/asv/app_suite.py

```diff
@@ -44,3 +43,29 @@ def time_minvertex(self, n_layers):
         """Time a QAOA algorithm for finding the minimum vertex cover of a graph."""
         hyperparams = {'n_layers': n_layers}
         benchmark_qaoa(hyperparams)
+
+    def peakmem_minvertex(self, n_layers):
+        """Benchmark the peak memory usage of QAOA algorithm for finding the minimum vertex cover of
+        a graph."""
+        hyperparams = {'n_layers': n_layers}
+        benchmark_qaoa(hyperparams)
+
+class QAOA_heavy:
+    """Benchmark the QAOA algorithm for finding the minimum vertex cover of a large graph using
+    a large number of layers."""
+
+    n_layers = 5
+    graph = nx.complete_graph(20)
```

Comment:
@soranjh Could you please comment a bit more on how you arrived at these numbers? 20 qubits seems like quite a lot.

Also, why use the complete graph? It's an interesting choice. Min vertex cover is trivial there, it's just all the vertices. But then again the unconstrained cost Hamiltonian becomes more complicated

Reply:
20 nodes is the largest number we can have without getting the benchmark timed out error. The time out default is currently 60 s. The combination of 5 layers / 20 nodes is the heaviest benchmark we can have within the 60 s limit. The choice of the complete graph is arbitrary and we can define any graph we want.